art/attacks/evasion/simba.py

```diff
@@ -0,0 +1,319 @@
+# MIT License
+#
+# Copyright (C) IBM Corporation 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""
+This module implements the black-box attack `simba`.
+
+| Paper link: https://arxiv.org/abs/1905.07121
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+import logging
+
+import numpy as np
+from scipy.fftpack import dct, idct
+
+from art.attacks.attack import EvasionAttack
+from art.classifiers.classifier import ClassifierGradients
+from art.config import ART_NUMPY_DTYPE
+from art.utils import compute_success
+
+logger = logging.getLogger(__name__)
+
+
+class SimBA(EvasionAttack):
+    attack_params = EvasionAttack.attack_params + ['attack', 'max_iter', 'epsilon', 'order', 'freq_dim', 'stride', 'targeted', 'batch_size',]
+
+    def __init__(self, classifier, attack='dct', max_iter=3000, order='random', epsilon=0.1, freq_dim=4, stride=1, targeted=False, batch_size=1):
+        """
+        Create a SimBA (dct) attack instance.
+
+        :param classifier: A trained classifier.
+        :type classifier: :class:`.Classifier`
+        :param attack: attack type: pixel (px) or DCT (dct) attacks
+        :type attack: `str`
+        :param max_iter: The maximum number of iterations.
+        :type max_iter: `int`
+        :param epsilon: Overshoot parameter.
+        :type epsilon: `float`
+        :param order: order of pixel attacks: random or diagonal (diag)
+        :type order: `str`
+        :param freq_dim: dimensionality of 2D frequency space (DCT).
+        :type freq_dim: `int`
+        :param stride: stride for block order (DCT).
+        :type stride: `int`
+        :param targeted: perform targeted attack
+        :type targeted: `bool`
+        :param batch_size: Batch size (but, batch process unavailable in this implementation)
+        :type batch_size: `int`
+        """
+        super(SimBA, self).__init__(classifier=classifier)
```

Comment:
The keyword change to `estimator` in ART 1.3.
```suggestion
        super(SimBA, self).__init__(estimator=classifier)
```